src/Libraries/Microsoft.Extensions.Caching.Hybrid/HybridCacheBuilderExtensions.cs

```diff
@@ -59,4 +61,27 @@ public static IHybridCacheBuilder AddSerializerFactory<
         _ = Throw.IfNull(builder).Services.AddSingleton<IHybridCacheSerializerFactory, TImplementation>();
         return builder;
     }
+
+    /// <summary>
+    /// Register a default <see cref="JsonSerializerOptions"/> for use with JSON serialization.
+    /// </summary>
+    /// <returns>The <see cref="IHybridCacheBuilder"/> instance.</returns>
+    [Experimental(DiagnosticIds.Experiments.HybridCache, UrlFormat = DiagnosticIds.UrlFormat)]
+    public static IHybridCacheBuilder WithJsonSerializerOptions(this IHybridCacheBuilder builder, JsonSerializerOptions options)
```

Comment:
This API and design is a little different from how we do it elsewhere (minimal APIs and SignalR). Instead of giving a full JsonSerializerOptions, the developer "configures" a single JsonSerializerOptions. I guess the case the HybridCache covers is if you want a completely different set of options for a specific type.

See 

https://github.com/dotnet/aspnetcore/blob/aebd07af9854541970d0c05d99dc2e6e8bd2fdd4/src/ProjectTemplates/Web.ProjectTemplates/content/WebApiAot-CSharp/Program.cs#L6-L9

for an example at how people configure the JsonSerializerOptions for minimal APIs.

cc @halter73 @BrennanConroy

Reply:
Happy to refactor to any preferred pattern

Reply:
>  if you want a completely different set of options for a specific type

Is that a hard requirement? It doesn't seem to be possible minimal APIs and SignalR today, AFAICT.

Not that I'm against this design, if it is necessary. But I guess I would have thought we would follow the same design/pattern as the other 2 places that do this.

cc @captainsafia as well

Reply:
It looks like the overall aim could be met by inserting multiple type handlers to a single options, but: I'll check

Reply:
Talking it over with @captainsafia, since you could return an entirely different SerializerFactory per Type (i.e. use Json for some types, and XML for others), having this design / API allowing for different JsonSerializerOptions per Type seems OK.